gui/src/main.rs

```diff
@@ -242,6 +239,41 @@ impl App for Gui {
                 }
                 ActiveWindow::HISTORY => {
                     ui.label("History");
+                    let history_items_clone = Rc::clone(&self.request_history_items);
+                    let history_items = history_items_clone.borrow();
+                    if let Some(item_vec) = &*history_items {
+                        for item in item_vec {
+                            if ui.selectable_value(
+                                &mut self.selected_history_item,
+                                Rc::new(RefCell::from(Some(item.clone()))),
+                                format!("{}", item.id)
+                            ).clicked() {
+                                // TODO - replace url, method, request body, response body
+                                let requests_clone = self.saved_requests.borrow();
+                                let responses_clone = self.saved_responses.borrow();
+                                let requests = requests_clone.as_ref().unwrap();
+                                let responses = responses_clone.as_ref().unwrap();
+                                let historical_request = requests.get(&item.request_id).unwrap();
+                                let historical_response = responses.get(&item.response_id).unwrap();
+                                self.url = historical_request.url.clone();
+                                self.selected_http_method = HttpMethod::from_str(&historical_request.method).unwrap();
+                                match &historical_request.body {
+                                    Some(body_json) => {
+                                        let body_str = serde_json::from_value(body_json.clone()).unwrap();
+                                        self.body_str = body_str;
+                                    },
+                                    None => self.body_str = String::from(""),
+                                }
+                                let ui_response_clone = self.response.clone();
+                                let mut ui_response_guard = ui_response_clone.try_write().unwrap();
+                                let response_body = &historical_response.body;
+                                match response_body {
+                                    Some(body) => *ui_response_guard = Some(body.clone()),
```

Comment:
Since this a RwLock and only one thing has permission to update at a time we need to request permission to write, which thens provides a pointer to the memory, then use the * dereference operator to update that value.